fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+        files"
+        assert files_dir is not None, "The param files_dir should points to parent directory of gRPC interface files"
+
+        if interface_files:
+            abs_file_path_json = dict()
+            interface_module = []
+
+            # creating files abs path
+            for file_name in interface_files:
+                abs_file_path = os.path.join(base_dir, files_dir, file_name)
+                abs_file_path_json[file_name] = abs_file_path
+
+            # Adding Module path into sys.path variable
+            interface_dir_path = os.path.join(base_dir, files_dir)
+            if interface_dir_path not in sys.path:
+                sys.path.append(interface_dir_path)
+
+            # compatibility for python 2.7 and 3.6
+            logging.info("Loading The Modules : {}".format(interface_files))
+
+            try:
+                import importlib.util
+                for file_name in interface_files:
+                    if file_name.endswith('.py'):
+                        module_name = file_name.split('.')[0]
+                        spec = importlib.util.spec_from_file_location(module_name, abs_file_path_json[file_name])
+                        if spec:
+                            module_obj = importlib.util.module_from_spec(spec)
+                            spec.loader.exec_module(module_obj)
+                            interface_module.append(module_obj)
+                    else:
+                        logging.debug("No python modules found to import.")
+
+            except ImportError as e:
+                logging.debug("The Import Error raised for the module importlib trying with imp module : {}".format(e))
+                import imp
+                for file_name in interface_files:
+                    if file_name.endswith('.py'):
+                        module_name = file_name.split('.')[0]
+                        module_obj = imp.load_source(module_name, abs_file_path_json[file_name])
+                        interface_module.append(module_obj)
+            return interface_module
+        else:
+            logging.debug("No Interface file found, please specify the interface file")
+            return None
+
+    def _get_grpc_class(self, grpc_module_list, class_name):
+        """
+        component that return the given class object from the list of given module objects,
+        evaluates based on the given class names.
+        :param grpc_module_list: imported interface pb2 files module objects list
+        :param class_name:  interested class name in the given module objects list
+        :return:  tuple of name and class object of the given class name if class name found in the given module
+        objects, else returns an empty tuple.
+        """
+        for grpc_module in grpc_module_list:
+            for name, obj in inspect.getmembers(grpc_module, inspect.isclass):
+                logging.debug("Class Name : {} Object : {}".format(name, obj))
```

Comment:
* Don't use `logging` like this directly. Put `log = logging.getLogger(__name__)` at the top of the file and only ever use `log.info, log.debug, log.error`

Reply:
ok